### PR TITLE
HistoricOneToOneField

### PR DESCRIPTION
**Pull request by JordanHyatt**
Adds support for HistoricOneToOneField

## Description
Implemented the solution proposed by @quadracik  in issue  #1031 

## Related Issue
This PR addresses issue #1031 and #1389 

## How Has This Been Tested?
Appropriate unit tests have been written and follow the ones written for HistoricForeignKey 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have run the `pre-commit run` command to format and lint.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] I have added my name and/or github handle to `AUTHORS.rst`
- [x] I have added my change to `CHANGES.rst`
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by kgeidel — APPROVED** [review]

**Comment by erry-whiskypartners** [comment]
Hi, is this going to be merged and if so when please

**Comment by JordanHyatt** [comment]
> Hi, is this going to be merged and if so when please

Waiting for maintainers to review!

File: simple_history/tests/models.py
Comment (tim-schilling) [inline simple_history/tests/models.py:998]:
```suggestion
    In this case it should simply behave like OneToOneField because
    the origin model (this one) cannot be historic, so one to one field
    lookups are always "current".
```

File: simple_history/tests/models.py
Comment (tim-schilling) [inline simple_history/tests/models.py:1013]:
```suggestion
    In this case it should simply behave like OneToOneField because
    the origin model (this one) can be historic but the target model
    is not, so one to one field lookups are always "current".
```

File: simple_history/tests/models.py
Comment (tim-schilling) [inline simple_history/tests/models.py:1025]:
```suggestion
    Historic table with one to one relationship to historic table.
```

File: simple_history/tests/models.py
Comment (tim-schilling) [inline simple_history/tests/models.py:1031]:
```suggestion
    In this case as_of queries on the origin model (this one)
    or on the target model (the other one) will traverse the
    one to one field relationship honoring the timepoint of the
    original query.  This only happens when both tables involved
    are historic.
```

File: simple_history/tests/models.py
Comment (tim-schilling) [inline simple_history/tests/models.py:1035]:
```suggestion
    NOTE: related_name has to be different than the one used in
          TestParticipantToHistoricOrganizationOneToOne as they are
          sharing the same target table.
```

File: simple_history/tests/tests/test_models.py
Comment (tim-schilling) [inline simple_history/tests/tests/test_models.py:2857]: Are these comments referring to foreign key correct? Should any be reworded to one to one field?

File: docs/signals.rst
Comment (tim-schilling) [inline docs/signals.rst:33]: I think this was a rebasing error

File: docs/querying_history.rst
Comment (tim-schilling) [inline docs/querying_history.rst:168]: I think we can improve the usability of the docs by changing this to

```rst

.. _`HistoricForeignKey`:

HistoricForeignKey
------------------

...

HistoricOneToOneField
---------------------

Similar to :ref:`HistoricForeignKey`, but for OneToOneFields instead.
```

**Comment by tim-schilling** [comment]
We also need to credit @david-homelend in the commit. See https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors

File: simple_history/models.py
Comment (JordanHyatt) [reply simple_history/models.py:944]: For sure, I'm all about keeping it DRY.

File: simple_history/models.py
Comment (JordanHyatt) [reply simple_history/models.py:972]: thank you for the doc string!

File: simple_history/models.py
Comment (JordanHyatt) [reply simple_history/models.py:944]: By implementing a get_related_model method I was able to DRY out all 3.  Let me know what you  think

File: simple_history/tests/tests/test_models.py
Comment (JordanHyatt) [reply simple_history/tests/tests/test_models.py:2857]: They should be

**Comment by JordanHyatt** [comment]
Ok, I made the requested changes,  let me know if I missed/messed something up!

**Comment by JordanHyatt** [comment]
I had to update language_version: python3.11  in the .pre-commit-config.yaml  because it was failing to find a python3.8 interpreter

File: simple_history/models.py
Comment (tim-schilling) [reply simple_history/models.py]: Looks good to me!

**Review by tim-schilling — APPROVED** [review]
I made some last clean-up tweaks. When CI passes, I'll merge. Thank you for the PR!

**Comment by JordanHyatt** [comment]
> I made some last clean-up tweaks. When CI passes, I'll merge. Thank you for the PR!

Sounds Good!  I think we also need to give [quadracik](https://github.com/quadracik) credit.   It was his solution from issue #1031  that I mostly borrowed from.

**Comment by tim-schilling** [comment]
It worked! :tada:

**Comment by quadracik** [comment]
Thanks guys! I really appreciate it.

**Pull request merged**
